superset-frontend/src/components/Chart/Chart.tsx

```diff
@@ -178,6 +182,17 @@ class Chart extends PureComponent<ChartProps, {}> {
   renderStartTime: any;
 
   constructor(props: ChartProps) {
+    if (props.datasource?.metrics) {
+      // eslint-disable-next-line no-param-reassign
+      props.datasource.currencyFormats = Object.fromEntries(
+        props.datasource.metrics
+          .filter(
+            (metric): metric is Metric & { currency: Currency } =>
+              metric.currency !== undefined,
+          )
+          .map(metric => [metric.metric_name, metric.currency]),
+      ) as Record<string, Currency>;
+    }
```

Comment:
I know that modifying `chartProps` here at this stage is not the most elegant solution, but I think it's the DRYer way. If we don't do it here, then in every `transformProps` we would have to add similar logic (or call a util that does this).

Anyways, open to feedback! There could be another option I couldn't find.

Reply:
that's a great suggestion! Working on that now